src/dmd/expressionsem.d

```diff
@@ -11508,3 +11484,55 @@ private bool checkFunctionAttributes(Expression exp, Scope* sc, FuncDeclaration
         return error;
     }
 }
+
+/*******************************
+ * Helper function for `getRightThis()`.
+ * Gets `this` of the next outer aggregate.
+ * Params:
+ *      loc = location to use for error messages
+ *      sc = context
+ *      s = the parent symbol of the existing `this`
+ *      ad = struct or class we need the correct `this` for
+ *      e1 = existing `this`
+ *      t = type of the existing `this`
+ *      var = the specific member of ad we're accessing
+ *      flag = if true, return `null` instead of throwing an error
+ * Returns:
+ *      Expression representing the `this` for the var
+ */
+Expression skipNestedFunctions(const ref Loc loc, Scope* sc, Dsymbol s, AggregateDeclaration ad, Expression e1, Type t, Dsymbol var, bool flag = false)
```

Comment:
The fact that this name does not have `getThis` in it makes it less obvious when trying to comprehend the usage sites.  What about `getThisSkippingNested` or something like that?

Reply:
agreed

Reply:
renmed to `getThisSkipNestedFuncs`.